controller/apps_osioclient.go

```diff
@@ -62,54 +62,54 @@ func (osioclient *OSIOClientV1) GetNamespaceByType(ctx context.Context, userServ
 func (osioclient *OSIOClientV1) GetUserServices(ctx context.Context) (*app.UserService, error) {
 	resp, err := osioclient.wc.ShowUserService(ctx, witclient.ShowUserServicePath())
 	if err != nil {
-		return nil, err
+		return nil, errs.Wrapf(err, "could not retrieve uses services")
 	}
 
 	defer resp.Body.Close()
 
 	respBody, err := ioutil.ReadAll(resp.Body)
 
 	status := resp.StatusCode
-	if status == 404 {
+	if status == http.StatusNotFound {
 		return nil, nil
 	} else if status < 200 || status > 300 {
-		return nil, errors.New("Failed to GET " + witclient.ShowUserServicePath() + " due to status code " + string(status))
+		return nil, errs.Errorf("failed to GET %s due to status code %d", witclient.ShowUserServicePath(), status)
 	}
 
 	var respType app.UserServiceSingle
 	err = json.Unmarshal(respBody, &respType)
 	if err != nil {
-		return nil, err
+		return nil, errs.Wrapf(err, "could not unmarshal user services JSON")
 	}
 	return respType.Data, nil
 }
 
 // GetSpaceByID - fetch space given UUID
 func (osioclient *OSIOClientV1) GetSpaceByID(ctx context.Context, spaceID uuid.UUID) (*app.Space, error) {
 	// there are two different uuid packages at play here:
-	//   github.com/satori/go.uuid and goadesign/goa/uuid.
-	// because of that, we fenerate our own URL to avoid issues for now.
+	// github.com/satori/go.uuid and goadesign/goa/uuid.
+	// because of that, we generate our own URL to avoid issues for now.
 	urlpath := fmt.Sprintf("/api/spaces/%s", spaceID.String())
```

Comment:
use app.ShowSpacePath()

Reply:
The code looks horrible since I can't just use one type in place of another like in C++:
	var guid goauuid.UUID
	for i, b := range spaceID.Bytes() {
		guid[i] = b
	}
	urlpath := witclient.ShowSpacePath(guid)
I really wish Goa didn't have its own UUID type.

Reply:
This doesn't work? They are both just [16]byte types
```go
witclient.ShowSpacePath(guuid.UUID(spaceID))
```

Reply:
That works perfectly.  My inexperience with Go shows, I was trying various other syntax that didn't (like assignment).  Thanks.  It baffled me how there was no way to assign one array of 16 bytes to another, even if their types were differently named.